packages/diagrams/frontend/sirius-components-diagrams/src/renderer/reconnect-edge/useReconnectEdge.tsx

```diff
@@ -113,7 +114,8 @@ export const useReconnectEdge = (): UseReconnectEdge => {
           reconnectEdgeKind = GQLReconnectKind.SOURCE;
           newEdgeTargetId = newConnection.source;
         }
-        if (newEdgeTargetId) {
+        const newTargetNode = store.getState().nodeLookup.get(newEdgeTargetId);
+        if (newEdgeTargetId && newTargetNode && !isIconLabelNode(newTargetNode)) {
```

Comment:
- We want to remove `IconLabelNode.tsx` in favor of only have `FreeFormNode` and `ListNode`
- I don't see why any feature should be specific to the appearance of a node